### PR TITLE
[WIP] Feature: notification via kafka

### PR DESCRIPTION
**Pull request by goern**
Depends-On: thoth-station/python#160

This will run as a background task on OpenShift, and wake up periodically to pull in updates from Python Package Indices known to thoth. And update will be pushed as a PackageRelease to a Kafka topic, no webhooks are sent out.

### CODE REVIEW

File: dumper.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+_KAFKA_TOPIC_RETENTION_TIME_SECONDS = 60 * 60 * 24 * 45
+
+
+ssl_context = ssl.create_default_context(purpose=ssl.Purpose.SERVER_AUTH, cafile=_KAFKA_CAFILE)
+app = faust.App("thoth_package_releases", broker=_KAFKA_BOOTSTRAP_SERVERS, ssl_context=ssl_context, web_enabled=False)
+package_releases_topic = app.topic(
+    THOTH_PACKAGE_RELEASES_TOPIC_NAME,
+    value_type=PackageRelease,
+    retention=_KAFKA_TOPIC_RETENTION_TIME_SECONDS,
+    partitions=1,
```
Comment (KPostOffice) [inline dumper.py]: `internal=True` to designate it as a Faust topic

File: pull_updates.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                        package_name,
+                        package_version,
+                        package_index.url,
+                    )
+
+                    await package_releases_topic.send(
+                        value=PackageRelease(package_index.url, package_name, package_version)
+                    )
+                else:
+                    _LOGGER.debug(
```
Comment (KPostOffice) [inline pull_updates.py]: `app.logger.debug`

File: dumper.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+from thoth.messaging import PackageRelease, THOTH_PACKAGE_RELEASES_TOPIC_NAME
+
+
+__version__ = "0.1.0-dev"
+
+_DEBUG = os.getenv("DEBUG", False)
+
+
+init_logging()
+_LOGGER = logging.getLogger("dumper")
```
Comment (KPostOffice) [inline dumper.py]: With faust apps you have to use app.logger, like this 
```python
app.log.debug("Failed to publish with the following error message: %r", identifier.msg)
```

File: pull_updates.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+
+@app.task()
+async def connect_graph():
+    """Connect to Thoth's Knowledge Graph."""
+    global graph
+
+    graph = GraphDatabase()
+    graph.connect()
+    _LOGGER.info(f"connected to Thoth's Knowledge Graph using thoth.storages v{thoth_storages_version} ...")
```
Comment (fridex) [inline pull_updates.py]:
```
    _LOGGER.info(f"connected to Thoth's Knowledge Graph using thoth.storages v%s... ", thoth_storages_version)
```

File: pull_updates.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+_LOGGER = logging.getLogger("pull_updates")
+_LOGGER.setLevel(logging.DEBUG if _DEBUG else logging.INFO)
+
+_KAFKA_BOOTSTRAP_SERVERS = os.getenv("KAFKA_BOOTSTRAP_SERVERS", "localhost:9092")
+_KAFKA_CAFILE = os.getenv("KAFKA_CAFILE", "secrets/data-hub-kafka-ca.crt")
+_KAFKA_TOPIC_RETENTION_TIME_SECONDS = 60 * 60 * 24 * 45  # seconds
+_PULL_UPDATES_TIMER_SLEEP_SECONDS = os.getenv("PULL_UPDATES_TIMER_SLEEP", 60 * 5)
+
+
+graph = None
```
Comment (fridex) [inline pull_updates.py]:
```
_GRAPH
```

File: dumper.py (hunk 3 above)
Comment (fridex) [reply dumper.py]: Is there any faust feature enforcing this? Generally this depends on the logger you want to use to log your messages.

The logger for all Thoth components/libraries live in "thoth" namespace, this enables as to transparently configure logging facilities across the whole application (see thoth-common readme).

The logger should be named:

```
_LOGGER = logging.getLogger("thoth.dumper")
```

**Comment by fridex** [comment]
> The only issue that I see other than logging statements is that these don't run in the s2i configuration. Either another option needs to be added to app.py cli which will do something like `subprocess.call("faust -l debug -A pull_updates")` unless app.py is being replaced by this implementation in which case it might make more sense to create a shell script.

Please no subprocess. You can configure loglevel for any logger:

https://docs.python.org/3/library/logging.html#logging.Logger.setLevel

**Comment by pacospace** [comment]
I guess this can be closed @goern !

**Pull request closed without merging**